test/extended/util/client.go

```diff
@@ -111,6 +112,57 @@ type WorkSpace struct {
 	ParentServerURL string // WorkSpace ParentServerURL E.g. https://{{kcp-service-domain}}/clusters/root:orgID
 }
 
+var (
+	loadConfigOnce sync.Once
+	testContext    string
+	orgServer      string
+	homeServer     string
+)
+
+// loadConfig get the User orgnaization workspace and home workspace server
+func loadConfig() {
+	var err error
+	// If not set "E2E_TEST_CONTEXT" use "kcp-stable" testContext by default
+	testContext = os.Getenv("E2E_TEST_CONTEXT")
+	if testContext == "" {
+		testContext = "kcp-stable"
+		e2e.Debugf(`Env var "E2E_TEST_CONTEXT" does not exist, using kcp-stable context`)
+	}
+	configJSON := ReadKubeConfig(KubeConfigPath())
+	orgServer = gjson.Get(configJSON, `clusters.#(name="`+testContext+`").cluster.server`).String()
+	e2e.Debugf(`User orgnaization workspace server is: "%s"`, orgServer)
+	client := &CLI{
+		username:        "admin",
+		execPath:        "kubectl",
+		adminConfigPath: KubeConfigPath(),
+	}
+	rootServer := GetParentWsServerURL(orgServer)
+	homeServer, err = client.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().NotShowInfo().Run("get").Args("workspace/~", "--server="+rootServer, "-o=jsonpath={.status.URL}").Output()
```

Comment:
WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().NotShowInfo() looks too long, must they be used? I did not yet check deep.

Reply:
Yeah. it must be set. I change to another way to make it better. Fixed.